src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -0,0 +1,366 @@
+"""
+    SemidiscretizationCoupled
+
+A struct used to bundle multiple semidiscretizations.
+`semidiscretize` will return an `ODEproblem` that synchronizes time steps between the semidiscretizations.
+Each call of `rhs!` will call `rhs!` for each semidiscretization individually.
+The semidiscretizations can be coupled by gluing meshes together using `BoundaryConditionCoupled`.
+
+!!! warning "Experimental code"
+    This is an experimental feature and can change any time.
+"""
+struct SemidiscretizationCoupled{S, I} <: AbstractSemidiscretization
+  semis::S
+  u_indices::I # u_ode[u_indices[i]] is the part of u_ode corresponding to semis[i]
+  performance_counter::PerformanceCounter
+end
+
+
+"""
+    SemidiscretizationCoupled(semis)
+
+Create a coupled semidiscretization that consists of the semidiscretizations contained in the tuple `semis`.
+"""
+function SemidiscretizationCoupled(semis)
+  @assert all(semi -> ndims(semi) == ndims(semis[1]), semis) "All semidiscretizations must have the same dimension!"
+
+  # Number of coefficients as Vector
+  n_coeffs = semis .|> mesh_equations_solver_cache .|> (x -> n_coefficients(x...)) |> collect
```

Comment:
If this is the only place where you use `n_coefficients`, can't you just use `ndofs` here instead?